pkg/gpu/ebpf/c/gpu.c

```diff
@@ -29,6 +30,14 @@ static inline void load_dim3(__u64 xy, __u64 z, dim3 *dst) {
     dst->z = z;
 }
 
+static inline void fill_header(cuda_event_header_t *header, __u64 stream_id, cuda_event_type_t type) {
+    header->pid_tgid = bpf_get_current_pid_tgid();
+    header->ktime_ns = bpf_ktime_get_ns();
+    header->stream_id = stream_id;
+    header->type = type;
+    get_cgroup_name(header->cgroup, sizeof(header->cgroup));
```

Comment:
is there any kernel version limitations to use this function?

Reply:
The function itself has a check with `bpf_helper_exists` for `bpf_get_current_task`  so if it's not supported we'll get an empty container. However, `bpf_get_current_task` was added in 4.8.